_posts/2023-09-13-prospero-add-feature-pack.adoc

```diff
@@ -0,0 +1,149 @@
+---
+layout: post
+title:  "Installing additional feature packs in WIldfly"
+date:   2023-09-13
+tags:   prospero galleon wildfly
+author: spyrkob
+description: Using Prospero to installing additional feature packs in existing Wildfly servers.
+---
+
+== Additional feature packs for Wildfly
+
+Wildfly server capabilities can be extended by installing additional feature packs such as wildfly-datasources-galleon-pack or wildfly-myfaces-galleon-pack.
+The feature packs can be installed using Galleon or Prospero.
+
+In this article we will use Prospero to install a MariaDB datasource provided by wildfly-datasources-galleon-pack.
+
+== Prerequisites
+
+1. Download and unzip Wildfly 29.0.1.Final from https://www.wildfly.org/downloads/
+1. Checkout Prospero from https://github.com/wildfly-extras/prospero
+1. Build Prospero by running `mvn install -Pdist`
+
+== Registering Wildfly server with Prospero
+
+The first step to install additional feature packs in an existing Wildfly server, is to subscribe the server to wildfly channels.
+This allows Prospero to apply updates for Wildfly and modify the installation.
+
+[,bash]
+----
+$ ./prospero.sh update subscribe --dir /tmp/wildfly --product wildfly --version 29.0.1.Final
+----
+
+The `update subscribe` command generates a manifest based on selected version of Wildfly server and registers the server to a channel using this manifest:
+
+[,bash]
+----
+./prospero.sh channel list --dir /tmp/wildfly
```

Comment:
here again, if you could split the command to run `prospero.sh channel list --dir /tmp/wildfly` from its output , the user could directly paste it. THat's a pattern we should do in our code snippet.

```
You can run the `update subscribe` command to generate a manifest based on selected version of WildFly server and register the server to a channel using this manifest:
[source]
----
./prospero.sh channel list --dir /tmp/wildfly
----

The command will output the channels that are the server is subscribed to:

[source]
----
Server /tmp/wildfly is subscribed to following channels:

---
# wildfly
  manifest: file:/tmp/wildfly-29.0.1.Final/.installation/manifest-wildfly-29.0.1.Final.yaml
  repositories:
    id: central
    url: https://repo1.maven.org/maven2/
    id: jboss-public
    url: https://repository.jboss.org/nexus/content/groups/public/
    id: mrrc
    url: https://maven.repository.redhat.com/ga/
---
----
```